lib/ssh_scan/scan_engine.rb

```diff
@@ -9,13 +10,16 @@ def scan(opts)
       targets = opts[:targets]
       port = opts[:port]
       policy = opts[:policy_file]
+      timeout = opts[:timeout]
 
       # Connect and get results (native)
       result = []
       targets.each_with_index do |target, index|
-        client = SSHScan::Client.new(target, port)
+        client = SSHScan::Client.new(target, port, timeout)
         client.connect()
-        result.push(client.get_kex_result())
+        kex_result = client.get_kex_result()
+        result.push(kex_result)
+        next if kex_result.include?(:error)
 
         # Connect and get results (Net-SSH)
         net_ssh_session = Net::SSH::Transport::Session.new(target, :port => port)
```

Comment:
Can we also add a connect timeout for this too?

We might also need to catch the net-ssh specific exception for connect timeouts and re-raise the custom exception you created